### ISSUE TITLE
Are the aggregations correct for all ERA5-Land variables?

### ISSUE THREAD

**Issue by jonschwenk** [kratzert/Caravan#13]
The ERA5-Land hourly dataset on GEE provides (somewhat confusingly) two types time-aggregated bands for some variables. For example, there is `total_precipitation` and `precipitation_hourly`. The definitions of these according to the Data Catalog are as follows:

`total_precipitation` : [skipping stuff here related to defining precipitation] This variable is accumulated from the beginning of the forecast time to the end of the forecast step. The units of precipitation are depth in meters. It is the depth the water would have if it were spread evenly over the grid box. 

`total_precipitation_hourly` : Same as 'total_precipitation' except not accumulated and only for the given forecast step.

Currently, Caravan code is using the (temporal) sum of `total_precipitation`. My reading of the above descriptions makes me think this might be an error. It's not completely clear to me what "end of forecast step" means (what is the time of a forecast step?), but if the forecast step were an hour, I don't see why the `_hourly` bands would need to exist. This [stackexchange post ](https://gis.stackexchange.com/a/442599/78446) touches on this but doesn't really clarify what "end of forecast step" means.

To check this, I put together a [basic GEE example ](https://code.earthengine.google.com/09cf6fe2116f2c5b485d71685c197e02) to compute the (spatially) average precip in mm from daymet, ERA5L-Hourly `total_precipitation`, and ERA5L-Hourly `total_precipitation_hourly`. You can see that `total_precipitation` vastly overestimates the daily value.

If this is indeed an error, it seems like there are two options: use the `x_hourly` band instead of just `x` for the affected variables, or use the `max()` operator. You can see from my GEE example that using the max() gets you much closer to the benchmark value, but still doesn't agree with the `x_hourly` version.

In short, it seems that bands with corresponding `_hourly` bands are _cumulative_ (relative to what, I'm not sure, maybe it resets at 00:00 each day), so taking their sum isn't what we want to do. I might be wrong here as I assume you'd probably notice this when you looked at Caravan precip time series and see that they're order-of-magnitude(s) larger than they should be?

**Comment by kratzert** [comment]
Short answer: Yes, the correct bands are used and we take care of the accumulation manually in the 2nd notebook.

Longer answer: `total_precipitation` is the original ERA5-Land band, while `precipitation_hourly` is a derived band from Earth Engine. Of course, we are interested in the "instantaneous" precipitation, not the accumulated one. However, at the time when we started this project, there was a bug on the EE side that led to unreasonable values in `precipitation_hourly` (in the order of millions/billions of meters of rain at individual time steps). Therefore, I decided to use only the original ERA5-Land bands and to de-accumulate the accumulated bands myself. This happens in the second notebook, once the spatially aggregated (still hourly) data from Earth Engine is downloaded to your local machine. The second notebook takes care of a) de-accumulating the accumulated bands to hourly "instantaneous" values, b) shifting the data to local time, c) computing daily averages, and d) merging the data with streamflow.

You can find further details in the second notebook and here is the function that is called to deaccumulated the ERA5-bands (https://github.com/kratzert/Caravan/blob/main/code/caravan_utils.py#L322

And yes, this also applies for other bands, see https://github.com/kratzert/Caravan/blob/main/code/caravan_utils.py#L341

**Comment by jonschwenk** [comment]
Spend an hour opening an issue >> spend five minutes reading the code 😅 

Thanks for the detailed explanation--I knew something wasn't adding up since I didn't figure there was any way you would miss orders-of-magnitude error in precipitation. I didn't even think of it being as simple as `np.diff()` in post-processing.
